rewrite-gradle/src/main/java/org/openrewrite/gradle/trait/GradleDependency.java

```diff
@@ -68,6 +68,10 @@ public Matcher artifactId(String artifactId) {
                     return null;
                 }
 
+                if (!withinDependenciesBlock(cursor)) {
+                    return null;
+                }
+
                 GradleDependencyConfiguration configuration = gradleProject.getConfiguration(methodInvocation.getSimpleName());
                 if (configuration == null) {
```

Comment:
Let's here instead also allow `"classpath".equals(methodInvocation.getSimpleName())`. By doing this we can completely eliminate the `GradleDependencyPredicate` code altogether and only use the `GradleDependency.Matcher`.

Reply:
This is a good question. I decided to go ahead and try to correctly capture the `classpath` configuration via https://github.com/openrewrite/rewrite/pull/4459. If that PR gets merged, then we'll be able to check for the configuration within the buildscript's own configurations map and then get the `ResolvedDependency` just the same way as with a project dependency.

Reply:
Perfect. I will watch for your PR

Reply:
PRs are merged, so now you should be able to update the `GradleDependency` trait to look at the new `GradleBuildscript` in order to find the `classpath` configuration and from that the `ResolvedDependency` instance that will be available in that configuration.